### PR TITLE
[CI] Enable markdown linter and fix reported issues in `*.md`, `*/*.md`, and website/content

### LINKED ISSUE #168: [infra] Move cSpell config to a place where the VS Code plugin can pick it up automatically

### PR DESCRIPTION
**Pull request by chalin**
- Contributes to #156 by enabling markdown lint checks
- Fixes all issues reported by the markdown linter in `*.md`, `*/*.md`, and website/content files
- Closes #168
- Adds a top-level `package.json` file containing all packages needed to run checks on this repo. The `website/package.json` file contains packages (and scripts) for building the site. I'm not sure this split is the best, but it's worth a try
- **NOTE**: it's best to review changes by ignoring whitespace differences

### CODE REVIEW

File: governance/blogpost-process.md [hunk 1]
```diff
@@ -85,11 +85,11 @@ In the event of insufficient reviews, notify the assigned TAG Lead, seeking thei
...
 If there are open reviews that are not addressed, then the blog will not be published.
 Reviews are meant to provide valuable insight and suggestions for improving the quality of your content. By taking this perspective, you can view reviews as constructive opportunities for growth rather than personal criticism. By keeping an open mind and taking feedback into account, you can refine your blog post, amplify its impact, and provide a more engaging experience for your readers. Emphasizing the collaborative nature of reviews fosters a positive environment where continuous improvement is the driving force behind your writing efforts.
 
-If the review does not meet our [review guidelines](#reviewguidelines), please make this transparent in GitHub and give the reviewer the opportunity to self-improve in the way they conduct reviews. Focus on resolving a conflict before they occur.
+If the review does not meet our [review guidelines](#review-guidelines), please make this transparent in GitHub and give the reviewer the opportunity to self-improve in the way they conduct reviews. Focus on resolving a conflict before they occur.
```
Comment (chalin) [inline governance/blogpost-process.md:92]: Fixed broken anchor here.

File: governance/roles.md [hunk 2]
```diff
@@ -11,9 +11,9 @@ The following is the current listing of member roles:
 * [Chairs](#role-of-chairs)
 * [Technical Leads](#role-of-technical-leads)
 * [Project Leads](#role-of-project-leads)
-* [TAG Representatives](#role-of-tag-representatives)
-* [Team leads](#role-of-team-leads)
-* [Group Members](#role-of-members)
+* [TAG Representatives](#role-of-environmental-sustainability-tag-representatives)
```
Comment (chalin) [inline governance/roles.md:14]: Fixed broken anchor here.

File: governance/roles.md [hunk 3]
```diff
@@ -11,9 +11,9 @@ The following is the current listing of member roles:
...
+* Team leads
+* Group Members
```
Comment (chalin) [inline governance/roles.md:16]: There are no sections corresponding to "Team leads" and "Group members" AFAICT (and the anchors don't exist), so I removed the anchors.

**Comment by chalin** [comment]
All checks are green, and the new markdown check is running (15 sec) : https://github.com/cncf/tag-env-sustainability/actions/runs/5814640831/job/15764712297?pr=169

> image

**Review by cjyabraham — APPROVED** [review]
Looks good!

File: governance/blogpost-process.md (hunk 1 above)
Comment (leonardpahlke) [reply governance/blogpost-process.md:92]: Thanks

**Review by leonardpahlke — APPROVED** [review]
Thanks! LGTM

**Comment by leonardpahlke** [comment]
@chalin feel free to merge 🙌

**Pull request merged**